### DISCUSSION TITLE
Creating `ControlFlowGraph` from Javascript code using `biome_js_analyze` crate

### DISCUSSION THREAD

**Discussion by MichaelTheSynthCat** [biomejs/biome#4739] [Q&A] [answered]
Hi, I would like to use Biome toolchain to do static analysis of malicious Javascript code using Rust. I found that there is code for creating Control Flow Graphs from Javascript code but I don't how to use this implementation. I also tried `SemanticModel` from `biome_js_semantic` crate, but with this I had no problem because there is a simple function `biome_js_semantic::semantic_model`  that creates this model from provided Javascript.

My question is how can I create this [ControlFlowGraph](https://github.com/biomejs/biome/blob/c2dda66cc7533019bbd6c76960fcd68cd94cf1be/crates/biome_js_analyze/src/services/control_flow.rs) from raw Javascript code using Rust?

Thanks for help!

**Comment by ematipico** [comment, marked as answer]
There hasn't been any use case where we needed to use the CFG outside of the analyzer (analyzer rules), so we didn't create any API for that.

You're more than welcome to send a PR for that
